src/main/java/io/supertokens/jwt/JWTSigningFunctions.java

```diff
@@ -0,0 +1,164 @@
+
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.supertokens.Main;
+import io.supertokens.jwt.exceptions.UnsupportedJWTSigningAlgorithmException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.jwt.JWTAsymmetricSigningKeyInfo;
+import io.supertokens.pluginInterface.jwt.JWTSigningKeyInfo;
+import io.supertokens.pluginInterface.jwt.exceptions.DuplicateKeyIdException;
+
+import java.security.*;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.*;
+
+public class JWTSigningFunctions {
+    /**
+     * Creates and returns a JWT string
+     * @param main
+     * @param algorithm The signing algorithm to use when creating the token. Refer to {@link JWTSigningKey.SupportedAlgorithms}
+     * @param payload JSON object containing user defined claims to be added to the JWT payload
+     * @param jwksDomain Used as the issuer in the JWT payload
+     * @param jwtValidity Used to set iat anf exp claims in the JWT payload
+     * @return String token
+     * @throws StorageQueryException If there is an error interacting with the database
+     * @throws StorageTransactionLogicException If there is an error interacting with the database
+     * @throws NoSuchAlgorithmException If there is an error when using Java's cryptography packages
+     * @throws InvalidKeySpecException If there is an error when using Java's cryptography packages
+     * @throws JWTCreationException If there is an error when creating JWTs
+     * @throws UnsupportedJWTSigningAlgorithmException If the algorithm provided does not match any of the supported algorithms
+     * @throws DuplicateKeyIdException If there is an error storing generated keys in storage
+     */
+    public static String createJWTToken(Main main, String algorithm, JsonObject payload, String jwksDomain, long jwtValidity)
+            throws StorageQueryException, StorageTransactionLogicException, NoSuchAlgorithmException, InvalidKeySpecException,
+            JWTCreationException, UnsupportedJWTSigningAlgorithmException, DuplicateKeyIdException {
+        // TODO: In the future we will have a way for the user to send a custom key id to use
+        JWTSigningKey.SupportedAlgorithms supportedAlgorithm;
+
+        try {
+            supportedAlgorithm = JWTSigningKey.SupportedAlgorithms.valueOf(algorithm);
+        } catch (IllegalArgumentException e) {
+            // If it enters this block then the string value provided does not match the algorithms we support
+            throw new UnsupportedJWTSigningAlgorithmException();
```

Comment:
Add a test to make sure it comes here as expected.

Reply:
Done